services/cd-service/pkg/history/history.go

```diff
@@ -19,14 +19,86 @@ package history
 import (
 	"fmt"
 	"strings"
+	"sync"
 
+	"github.com/hashicorp/golang-lru"
 	git "github.com/libgit2/git2go/v33"
 )
 
+type cacheNode struct {
+	mx        sync.Mutex
+	children  map[string]*cacheNode
+	changedAt *git.Commit
+	cost      uint
+}
+
+func (c *cacheNode) get(name string) *cacheNode {
```

Comment:
it seems more like 'put' 🤔

Reply:
Yes. Now with two days after writing it, I think the wording is not perfect. Also it's called `cacheNode` but it's actually a `resultNode` as we are not storing the result anywhere else.

Reply:
I'll change some wordings here.